src/serverlessFunction/functionModel.ts

```diff
@@ -123,13 +123,18 @@ export class ServerlessFunctionModel implements Disposable {
     private addWatchers() {
         if (workspace.workspaceFolders) {
             for (const workspaceFolder of workspace.workspaceFolders) {
-                this.watchers.push(
-                    fs.watch(workspaceFolder.uri.fsPath, (_event, filename) => {
-                        if (filename === 'func.yaml') {
-                            this.view.refresh();
-                        }
-                    }),
-                );
+                try {
+                    fs.accessSync(workspaceFolder.uri.fsPath);
+                    this.watchers.push(
+                        fs.watch(workspaceFolder.uri.fsPath, (_event, filename) => {
+                            if (filename === 'func.yaml') {
+                                this.view.refresh();
+                            }
+                        }),
+                    );
+                } catch (e) {
```

Comment:
This catch will also catch fs.watch errors in case watcher cannot be installed. This error is very common when running on linux. There should be warning for the user that automatic updates for view are not going to be triggered due to filesystem watcher install failure.

Reply:
Agree with @dgolovin here. We probably should show a warning message here.

Reply:
Okay. I moved installing the file watcher out of the try/catch, and added an error message if the workspace folder doesn't exist, with an explanation why we are trying to watch it.